src/content/docs/setup/configuration/cors-setup.mdx

```diff
@@ -0,0 +1,154 @@
+---
+title: CORS Setup
+description: Quick guide to configuring CORS headers for Adobe Commerce GraphQL endpoints.
+---
+
+import { Aside } from '@astrojs/starlight/components';
+import Tasks from '@components/Tasks.astro';
+import Task from '@components/Task.astro';
+import Link from '@components/Link.astro';
+
+<Aside type="caution" title="Production recommendation">
+CORS should be a last resort for production. Prefer same-origin delivery of the storefront and API (for example, via Fastly VCL) to avoid CORS entirely.
+</Aside>
+
+## Quick test: Is CORS working?
+
+If you already have CORS configured, test it by replacing `YOUR_STOREFRONT_URL` with your actual frontend URL:
+
+```bash
+curl -I -X OPTIONS https://your-commerce-backend.com/graphql \
+  -H "Origin: YOUR_STOREFRONT_URL" \
+  -H "Access-Control-Request-Method: POST"
+```
+
+Look for these headers in the response:
+- `Access-Control-Allow-Origin: YOUR_STOREFRONT_URL` (or `*` if using a wildcard)
+- `Access-Control-Allow-Methods: GET, POST, OPTIONS`
+
+If you see these headers with your storefront URL, CORS is configured correctly. If not, follow the setup steps below.
+
+## When do you need CORS?
+
+You need CORS when your storefront runs on a different domain or port than your Adobe Commerce backend (for example, `http://localhost:3000` → `https://commerce.example.com`).
+
+Without CORS, browsers block cross-origin requests and you'll see errors like:
+
+```txt
+Access to fetch at 'https://commerce.example.com/graphql' from origin 'http://localhost:3000' 
+has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present
+```
+
+## Quick steps
+
+<Tasks>
+
+<Task>
+
+### Add CORS support
+
+To add CORS headers to your Adobe Commerce GraphQL endpoints, you can configure your server, implement a custom module, or use a third-party CORS module (such as [graycore/magento2-cors](https://github.com/graycoreio/magento2-cors)).
+
+If using a CORS module, install it via Composer:
+
+```bash
+composer require <module-package-name>
+```
+
+Refer to your chosen module's documentation for specific installation and configuration steps.
+
+</Task>
+
+<Task>
+
+### Enable the module
+
+If using a CORS module, enable it and run setup commands:
+
+```bash
+php bin/magento module:enable <Module_Name>
```

Comment:
We generally leave out the `php` from the command line. I've seen other pages in this repo that also omit it, but I don't know what's standard here.

Reply:
If it's a general practice to leave it off, i'll do the same here.